crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -823,14 +824,58 @@ def evacuate_network_node(controller, network_node, delete_namespaces = false)
         )
       end
 
+      def prepare_all_compute_nodes
+        ["kvm", "xen"].each do |virt|
+          prepare_compute_nodes virt
+        end
+      end
+
+      # Prepare the compute nodes for upgrade by upgrading necessary packages
+      def prepare_compute_nodes(virt)
+        save_upgrade_state("Preparing compute nodes of #{virt} type for upgrade")
```

Comment:
I'd suggest ```"Preparing #{virt} compute nodes for upgrade"``` or  ```"Preparing compute nodes of type: #{virt} for upgrade"```. Though you might want to check back with a native speaker.

Similar it true for all the other place using ```"#{virt} type"```